### PR TITLE
TizenRT: update tests to run on TizenRT

### PR DESCRIPTION
**Pull request by kolipka**
- FS tests can run from ROM
- tests that hungs or core dump are temporarly skipped
- tests for not implemented features marked
- temporary fix for stat not supporting relative path

Current status:
PASS : 79
FAIL : 4
TIMEOUT : 0
SKIP : 39

IoT.js-DCO-1.0-Signed-off-by: Konrad Lipner k.lipner@samsung.com

### CODE REVIEW

File: src/js/module.js [hunk 1]
```diff
@@ -64,33 +64,43 @@ iotjs_module_t.resolveFilepath = function(id, directories) {
 
   for(var i = 0; i<directories.length ; i++) {
     var dir = directories[i];
+    var modulePath = dir + id;
+
+    // START: Temprorary fix for TizenRT
+    // See: https://github.com/Samsung/TizenRT/issues/320
+    if (process.platform === 'tizenrt') {
+      if (modulePath[0] !== '/') {
```
Comment (LaszloLango) [inline src/js/module.js]: Please merge the two `if` statements.

File: test/run_pass/test_fs_open_close.js [hunk 2]
```diff
@@ -14,134 +14,121 @@
...
 
-process.on('exit', function() {
-  assert.equal(fs_async_normal_ok, true);
-  assert.equal(fs_async_normal_ok2, true);
-  assert.equal(fs_async_open_not_exist_ok, true);
-  assert.equal(fs_async_close_bad_fd_ok, true);
-});
+fs.open(fileName, 'r', '4' , function(err, fd) {
+  assert.equal(err, null,
+                  'Async Normal2: Error opening ' + fileName + ': ' + err);
```
Comment (LaszloLango) [inline test/run_pass/test_fs_open_close.js]: wrong indentation

File: test/run_pass/test_fs_open_close.js [hunk 3]
```diff
@@ -14,134 +14,121 @@
...
 
+  fs.close(fd, function(err) {
+    assert.equal(err, null,
+                    'Async Normal2: Error closing ' + fileName + ': ' + err);
```
Comment (LaszloLango) [inline test/run_pass/test_fs_open_close.js]: ditto

File: test/run_pass/test_fs_open_close.js [hunk 4]
```diff
@@ -14,134 +14,121 @@
...
+    });
+});
 
-assert.throws (function() {
+assert.throws(function() {
    var fd = fs.openSync(null, 123);
- }, TypeError);
-assert.throws (function() {
+ }, TypeError, 'Calling fs.openSync with null file path');
```
Comment (LaszloLango) [inline test/run_pass/test_fs_open_close.js]: ditto

File: test/run_pass/test_fs_open_close.js [hunk 5]
```diff
@@ -14,134 +14,121 @@
...
 
 ('rs sr r+ rs+ sr+ a a+')
-  .split(' ').forEach(function (flag){
+  .split(' ').forEach(function (flag) {
     assert.doesNotThrow(function (){
-      var fd = fs.openSync(process.cwd() + '/resources/test2.txt', flag);
+      var fd = fs.openSync(readFile, flag);
       fs.closeSync(fd);
-    }, 'file could not be opened with ' + flag);
+    }, 'File could not be opened with flag:' + flag);
```
Comment (LaszloLango) [inline test/run_pass/test_fs_open_close.js]: Please put a space after the colon.

File: test/run_pass/test_fs_open_close.js [hunk 6]
```diff
@@ -14,134 +14,121 @@
...
+  .split(' ').forEach(function (flag) {
     assert.doesNotThrow(function(){
-      var file = process.cwd() + '/resources/TEMP' + flag + '.txt';
+      var file = testRoot + '/TEMP' + flag + '.txt';
       var fd = fs.openSync(file, flag);
-      fs.unlinkSync(file);
       fs.closeSync(fd);
-    }, 'file could not be opened with ' + flag);
+      fs.unlinkSync(file);
+    }, 'File could not be opened with flag:' + flag);
```
Comment (LaszloLango) [inline test/run_pass/test_fs_open_close.js]: ditto

File: test/run_pass/test_fs_rename.js [hunk 7]
```diff
@@ -20,15 +20,32 @@
...
-  assert.equal(fs.existsSync(file1), false);
-  assert.equal(fs.existsSync(file2), true);
+  assert.equal(err, null, 'Rename error:' + err);
+  assert.equal(fs.existsSync(file1), false, 'Source file exist after renaming');
+  assert.equal(fs.existsSync(file2), true,
+               'Destination file not exist after renaming');
   fs.rename(file2, file1, function(err) {
-    assert.equal(err, null);
-    console.log("Pass");
+    assert.equal(err, null, 'Renaming back error:' + err);
```
Comment (LaszloLango) [inline test/run_pass/test_fs_rename.js]: ditto

**Comment by kolipka** [comment]
@LaszloLango @glistening  fixed, but maybe this is a time to resume https://github.com/Samsung/iotjs/issues/631? I found that test files are mixing styles a lot. That could be fixed automatically.

**Pull request merged**